### PR TITLE
AWS: fix errorprone warnings in S3URI

### PR DESCRIPTION
**Pull request by jackye1995**
fix errorprone warnings in S3URI：

```
/iceberg/aws/src/main/java/org/apache/iceberg/aws/s3/S3URI.java:60: warning: [StringSplitter] String.split(String) has surprising behavior
    String [] schemeSplit = location.split(SCHEME_DELIM);
                                          ^
    (see https://errorprone.info/bugpattern/StringSplitter)
  Did you mean 'List<String> schemeSplit = Splitter.onPattern(SCHEME_DELIM).splitToList(location);'?
/iceberg/aws/src/main/java/org/apache/iceberg/aws/s3/S3URI.java:73: warning: [StringSplitter] String.split(String) has surprising behavior
    path = path.split(QUERY_DELIM)[0];
                     ^
    (see https://errorprone.info/bugpattern/StringSplitter)
  Did you mean 'path = Iterables.get(Splitter.onPattern(QUERY_DELIM).split(path), 0);'?
/iceberg/aws/src/main/java/org/apache/iceberg/aws/s3/S3URI.java:74: warning: [StringSplitter] String.split(String) has surprising behavior
    path = path.split(FRAGMENT_DELIM)[0];
                     ^
    (see https://errorprone.info/bugpattern/StringSplitter)
  Did you mean 'path = Iterables.get(Splitter.onPattern(FRAGMENT_DELIM).split(path), 0);'?
/iceberg/aws/src/main/java/org/apache/iceberg/aws/s3/S3URI.java:79: warning: [MissingSummary] A summary fragment is required; consider using the value of the @return block as a summary fragment instead.
    * @return S3 bucket
     ^
    (see http://google.github.io/styleguide/javaguide.html#s7.2-summary-fragment)
  Did you mean '*Returns s3 bucket.'?
/iceberg/aws/src/main/java/org/apache/iceberg/aws/s3/S3URI.java:86: warning: [MissingSummary] A summary fragment is required; consider using the value of the @return block as a summary fragment instead.
    * @return S3 key
     ^
    (see http://google.github.io/styleguide/javaguide.html#s7.2-summary-fragment)
  Did you mean '*Returns s3 key.'?
5 warnings
```

### CODE REVIEW

**Comment by jackye1995** [comment]
@danielcweeks could you take a look of this?

File: aws/src/main/java/org/apache/iceberg/aws/s3/S3URI.java [hunk 1]
```diff
@@ -70,19 +70,21 @@
...
     String path = authoritySplit[1];
-    path = path.split(QUERY_DELIM)[0];
-    path = path.split(FRAGMENT_DELIM)[0];
+    path = path.split(QUERY_DELIM, -1)[0];
+    path = path.split(FRAGMENT_DELIM, -1)[0];
     this.key = path;
   }
 
   /**
+   * S3 bucket name
```
Comment (RussellSpitzer) [inline aws/src/main/java/org/apache/iceberg/aws/s3/S3URI.java:79]: The recommendation for these is to go with 
/**
 * Return s3 bucket
 */

So you still have a one liner, but it's the summary only

Comment (RussellSpitzer) [reply aws/src/main/java/org/apache/iceberg/aws/s3/S3URI.java:79]: Example from elsewhere int he codebase
```java
  /**
   * Returns the id of the field that the metrics within this class are associated with.
   */
  public int id() {
    return id;
  ```

Comment (jackye1995) [reply aws/src/main/java/org/apache/iceberg/aws/s3/S3URI.java]: Thanks for the reference, updated!

**Comment by rdblue** [comment]
Thanks @jackye1995! Looks good to me.

**Pull request merged**